src/renderer/components/+workloads-pods/pod-details-secrets.tsx

```diff
@@ -19,27 +19,41 @@ export class PodDetailsSecrets extends Component<Props> {
   secretsLoader = autorun(async () => {
     const { pod } = this.props;
 
-    this.secrets = await Promise.all(
+    this.secrets = (await Promise.all(
       pod.getSecrets().map(secretName => secretsApi.get({
         name: secretName,
         namespace: pod.getNs(),
       }))
-    );
+    )).filter(Boolean);
   });
 
   render() {
+    const { pod } = this.props;
+
     return (
       <div className="PodDetailsSecrets">
         {
-          this.secrets.map(secret => {
-            return (
-              <Link key={secret.getId()} to={getDetailsUrl(secret.selfLink)}>
-                {secret.getName()}
-              </Link>
-            );
+          pod.getSecrets().map(secretName => {
+            const secret = this.secrets.find(secret => secret.getName() === secretName);
```

Comment:
Given that we are iterating through the pod's secrets to get its names. Should that imply that `PodDetailsSecrets.secrets` should be an `observable.map<string, Secret>()`?

Reply:
Refactored to use observable map instead.